design/pattern.go

```diff
@@ -4,7 +4,7 @@ const (
 	patternBase64      = `^[0-9a-zA-Z+/=]+$`
 	patternTimestamp   = `^[0-9]{1,18}$`
 	patternURLEncoded  = `^[0-9a-zA-Z-_.~]+$`
-	patternAccountUUID = `^[0-9a-f]{32}$`
-	patternDomainURI   = `^$` // TODO
+	patternAccountUUID = `^[0-9a-f-]{32}$`
+	patternDomainURI   = `^(?:[0-9a-zA-Z-._~]|%[0-9][0-9])+$` // following: RFC3986
```

Comment:
DomainURI removes from the new IrohaObject and DomainUUID is added. But, as a URI this is a correct URI pattern. 👍